### ISSUE TITLE
Dot register contains command letters

### ISSUE THREAD

**Issue by eugenk** [VSCodeVim/Vim#3773]
Issue Type: <b>Bug</b>

**Describe the bug**
The "." register contains command letters. It should, however, only contain what has been inserted previously.

**To Reproduce**

1.  Open a new document
2.  Press this key sequence: `itest<esc>o<C-r>.`
3.  What you see in the buffer is:
    ```
    test
    itest
    ```

**Expected behavior**
What you should see in the buffer:
```
test
test
```

Extension version: 1.8.0
VS Code version: Code 1.33.1 (51b0b28134d51361cf996d2f0a1c698247aeabd8, 2019-04-11T08:22:55.268Z)
OS version: Darwin x64 18.5.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz (8 x 2900)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.20GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

</details>

**Comment by J-Fields** [comment]
https://github.com/VSCodeVim/Vim/blob/master/src/mode/modeHandler.ts#L551
At this point, I don't see an easy way of getting the value we want. If we take only the `DocumentContentChangeAction` from `recordedState.actionsRun` we still have `<BS>` characters that we need to resolve. I actually don't even see `<CTRL>-<BS>` in the list at all when I do it...
Maybe as a start we could be content to improve matters without getting it right in all edge cases.

**Comment by danielo515** [comment]
The <c-r> in insert mode is not working at all for me. Is If supposed to work with any register?

**Comment by J-Fields** [comment]
@danielo515 Yeah `<C-r>a` should work in insert mode. Please submit a ticket if that's not the case for you

**Comment by danielo515** [comment]
> @danielo515 Yeah `<C-r>a` should work in insert mode. Please submit a ticket if that's not the case for you

Yes, I can confirm it works, it is just that I was not getting any visual clue like you do when you select a register. 
Thanks in any case

**Comment by J-Fields** [comment]
@danielo515 Sorry, could you clarify? Are you referring to the `"` that shows up in vim? That isn't currently implemented in insert mode.

**Comment by danielo515** [comment]
> @danielo515 Sorry, could you clarify? Are you referring to the `"` that shows up in vim? That isn't currently implemented in insert mode.

Yes, that is. But having something in the bottom bar could be good enough

**Comment by J-Fields** [comment]
Would you mind submitting a separate issue? That's a reasonable request but out of scope for this issue.
